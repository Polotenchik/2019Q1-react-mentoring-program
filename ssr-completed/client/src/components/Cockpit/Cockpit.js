import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import RadioButton from '../../common/Components/RadioButton/RadioButton';
import Arrow from '../../common/Components/Arrow/Arrow';
import Theme from '../../common/Theme/Theme';

const Cockpit = ({
  sortBy, sortOrder, toggleSortBy, toggleSortDirection, filmsCount,
}) => {
  return (
    <Background>
      <CockpitWrapper>
        <div>
          <span id="filmsCount">
            {filmsCount}
          </span>
          <span>
            {' movies found'}
          </span>
        </div>
        <SortWrapper>
          <form>
            <span>Sort by</span>
            <RadioButton
              inputName="sort"
              inputValue="release date"
              inputId="release_date"
              filtered={sortBy}
              handlerChange={toggleSortBy}
              transparent />
            <RadioButton
              inputName="sort"
              inputValue="rating"
              inputId="vote_average"
              filtered={sortBy}
              handlerChange={toggleSortBy}
              transparent />
          </form>
          <Arrow onClick={toggleSortDirection} down={sortOrder === 'desc'} />
        </SortWrapper>
      </CockpitWrapper>
    </Background>
  );
};

export default Cockpit;

Cockpit.propTypes = {
  sortBy: PropTypes.string,
  sortOrder: PropTypes.string,
  toggleSortBy: PropTypes.func,
  toggleSortDirection: PropTypes.func,
  filmsCount: PropTypes.number,
};

const Background = styled.div` 
    width: 100%;
    background-color: ${Theme.colors.grey};    
`;

const CockpitWrapper = styled.div`
    display: flex;
    justify-content: space-around;
    width: 80%;
    margin: 0 auto;
    padding: 10px 0;
`;

const SortWrapper = styled.div`
    display: flex;
`;
