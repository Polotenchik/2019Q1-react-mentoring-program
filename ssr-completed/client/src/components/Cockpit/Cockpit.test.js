import React from 'react';
import { shallow } from 'enzyme';
import Cockpit from './Cockpit';

describe('Cockpit Snapshot', () => {
    it('Cockpit Component should render correctly with given props', () => {
        const sortBy='release date';
        const toggleSortBy = jest.fn();
        const filmsCount = 10;
        const component = shallow(<Cockpit sortBy={sortBy} toggleSortBy={toggleSortBy} filmsCount={filmsCount} />);
        expect(component).toMatchSnapshot();
    });
});

