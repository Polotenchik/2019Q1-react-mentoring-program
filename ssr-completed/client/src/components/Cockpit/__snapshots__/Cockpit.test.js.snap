// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cockpit Snapshot Cockpit Component should render correctly with given props 1`] = `
<styled.div>
  <styled.div>
    <div>
      <span
        id="filmsCount"
      >
        10
      </span>
      <span>
         movies found
      </span>
    </div>
    <styled.div>
      <form>
        <span>
          Sort by
        </span>
        <RadioButton
          filtered="release date"
          handlerChange={[MockFunction]}
          inputId="release_date"
          inputName="sort"
          inputValue="release date"
          transparent={true}
        />
        <RadioButton
          filtered="release date"
          handlerChange={[MockFunction]}
          inputId="vote_average"
          inputName="sort"
          inputValue="rating"
          transparent={true}
        />
      </form>
      <Arrow
        down={false}
      />
    </styled.div>
  </styled.div>
</styled.div>
`;
